src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -398,31 +398,26 @@ private void AddCredentialToCache(Uri serverUri, CURLAUTH authAvail, NetworkCred
             }
         }
 
-        private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
+        private void AddResponseCookies(EasyRequest state, string cookieHeader)
         {
             if (!_useCookie)
             {
                 return;
             }
 
-            if (response.Headers.Contains(HttpKnownHeaderNames.SetCookie))
+            Uri serverUri = state._requestMessage.RequestUri;
```

Comment:
> You had mentioned that if we were to use the old url, on the second hop, a cookie would not be found for the new url and the test case would fail

Yes, @vijaykota.  You were right all along. I based my understanding on what was happening when the test case is running and my misreading of WinHttpHandler.